packages/common/src/compiler/exploreCompiler.ts

```diff
@@ -654,7 +654,10 @@ export class ExploreCompiler {
         const { refTable, refName } = getParsedReference(ref, currentTable);
 
         const referencedMetric = tables[refTable]?.metrics[refName];
-        if (referencedMetric === undefined) {
+        // This refName can reference a dimension when using custom SQL in DBT and searching for distinct values on searchFieldUniqueValues
+        const referencedDimension = tables[refTable]?.dimensions[refName];
+        const referencedField = referencedMetric || referencedDimension;
+        if (referencedField === undefined) {
```

Comment:
`compileMetricReference` should not compile a dimension reference. 
This bug originates because custom metrics were the only way to inject custom SQL into a queryMetric and the `field.sql` would most likely return the "raw" column. This means the bug also happens for dimensions with custom SQL with references. https://github.com/lightdash/lightdash/issues/8912

Since we have custom dimensions now, we can change the metric query defined in the project service to have something like:
```
        const distinctField: CustomSqlDimension = {
            id: `${field.name}_distinct`,
            type: CustomDimensionType.SQL,
            dimensionType: isDimension(field) ? field.type : DimensionType.STRING,
            name: `${field.name}_distinct`,
            table: field.table,
            sql: `DISTINCT ${field.sql}`,
        };
```

Reply:
yeah, that makes sense. I'll try using custom sql dimensions

Is there any reason (security?) why we build a fake metricQuery, rather than running the `select distinct from` directly against the warehouse ?

Reply:
An issue about using custom dimensions is that the distinct doesn't like the extra () around it. 

This doesn't work: 

![Screenshot from 2024-07-25 08-46-59](https://github.com/user-attachments/assets/c1514cbb-2f60-4f0a-b42b-ed4f93b30caa)

This works: 

![image](https://github.com/user-attachments/assets/d1509ff7-34ff-4ce0-a7b9-b38cd0e83a0e)

Reply:
I could run ' compileQuery' to ensure all permissions are in place (user attributes) and then run the distinct directly on the warehouse

Reply:
Let me know what you think of the new approach

Reply:
> Is there any reason (security?) why we build a fake metricQuery, rather than running the select distinct from directly against the warehouse ?

Was to make sure we generate the filters, joins, etc correctly.